content/en/js3/blocks/template-html/index.md

```diff
@@ -0,0 +1,69 @@
++++
+title = 'Simplifying element creation'
+headless = true
+time = 15
+facilitation = false
+emoji= '🧩'
+[objectives]
+    1='Use template tags to simplify element initialisation'
++++
+
+### Using `<template>` tags
+
+We could simplify this code with a different technique for creating elements.
+
+Until now, we have only seen one way to create elements: `document.createElement`. The DOM has another way of creating elements - we can _copy_ existing elements and then change them.
+
+HTML has a useful tag designed to help make this easy, [the `<template>` tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template). When you add a `<template>` element to a page, it doesn't get displayed when the page loads. It is an inert fragment of future HTML.
+
+We can copy any DOM node, not just `<template>` tags. For this problem, we will use a `<template>` tag because it is designed for this purpose.
+
+When we copy an element, its children get copied. This means we can write our template card as HTML:
+
+```html
+<template id="film-card">
+  <section>
+    <h3>Film title</h3>
+    <p>Director</p>
+    <time>Duration</time>
+    <data>Certificate</data>
+  </section>
+</template>
+```
+
+This is our template card. Place it in the body of your html. It doesn't show up! [Template HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template) is like a wireframe; it's just a _plan_. We can use this template to create a card for any film object. We will clone (copy) this template and populate it with data.
+
+```js
+const film = {
+  title: "Killing of Flower Moon",
+  director: "Martin Scorsese",
+  times: ["15:35"],
+  certificate: "15",
+  duration: 112,
+};
+
+const card = document.getElementById("film-card").content.cloneNode(true);
+// Now we are querying our cloned fragment, not the entire page.
+card.querySelector("h3").textContent = film.title;
+card.querySelector("p").textContent = `Director: ${film.director}`;
+card.querySelector("time").textContent = `${film.duration} minutes`;
+card.querySelector("data").textContent = film.certificate;
```

Comment:
Should a `data` tag not also have a `value` attribute? What's the advantage of using a `<data>` tag over a `<p>` tag here?

Reply:
Yeah it does. We often used data tags for certificates. Data is just like the time tag -- it gives you a human readable version and a structured data value, which is nice for thinking about different consumers of the same data.  I am torn here -- I just think the code samples are getting too long.